### DISCUSSION TITLE
How to configure mini.indentscope

### DISCUSSION THREAD

**Discussion by angelus2014** [nvim-mini/mini.nvim#130] [Q&A] [answered]
Hello,

I can't seem to figure out how to configure mini.indentscope properly, especially the 'animation' section. At the moment I have:
```
 65        require("mini.indentscope").setup({
    1         draw = {
    2           delay = 0,
⚠   3         ╎ animation = gen_animation('none'),
    4         },
```

But it doesn't work and the documentation doesn't help me much. I tried various different ways but all my attempts have been in vain. Could someone show me an example on how to set this up? Not just with 'none', but I mean in general, how to configure the animation.

Thank you in advance.

---
With regards,

Angelo Machils

**Comment by echasnovski** [comment, marked as answer]
> Hello,
> 
> I can't seem to figure out how to configure mini.indentscope properly, especially the 'animation' section. At the moment I have:
> 
> ```
>  65        require("mini.indentscope").setup({
>     1         draw = {
>     2           delay = 0,
> ⚠   3         ╎ animation = gen_animation('none'),
>     4         },
> ```
> 
> But it doesn't work and the documentation doesn't help me much. I tried various different ways but all my attempts have been in vain. Could someone show me an example on how to set this up? Not just with 'none', but I mean in general, how to configure the animation.
> 
> Thank you in advance.
> 
> With regards,
> 
> Angelo Machils

Hi!

You are close :) This is a working example:

```lua
require('mini.indentscope').setup({
  draw = {
    delay = 0,
    animation = require('mini.indentscope').gen_animation('none'),
  },
})
```

or slightly more tidy example with `'exponentialInOut'`:

```lua
local indentscope = require('mini.indentscope')
indentscope.setup({
  draw = {
    delay = 0,
    animation = indentscope.gen_animation('exponentialInOut'),
  },
})
```

The logic is as follows:
- `require()` call executes certain Lua source file ('lua/mini/indentscope.lua' in this case) and returns whatever it is said to return (table with methods in this case).
- The `setup()` call initializes module behavior (like make mappings, create autocommands, set config settings, etc.). This is done by using input `config` table which is **evaluated at the moment of table creation**.

Documentation usually assumes this kind of knowledge. But in this case there is [this note](https://github.com/echasnovski/mini.nvim/blob/7cdef92809c1e531d5f90b16834149a7e7f00de1/doc/mini-indentscope.txt#L118-L121) which might have given the clue. I'll see how it can be made better in this regard.

**Reply by angelus2014** [reply]
Owwwwww :)

Thank you, I have got it working as intended!


---
With regards,

Angelo Machils

**Reply by ngortheone** [reply]
Please add this 
```require('mini.indentscope').setup({
  draw = {
    delay = 0,
    animation = require('mini.indentscope').gen_animation('none'),
  },
})
```
to readme/docs, I found it a bit confusing to disable the animation and this answer helped!
Thanks

**Reply by echasnovski** [reply]
> Please add this
> 
> ```
>   draw = {
>     delay = 0,
>     animation = require('mini.indentscope').gen_animation('none'),
>   },
> })
> ```
> 
> to readme/docs, I found it a bit confusing to disable the animation and this answer helped! Thanks

It is present in the description of `config.draw.animation` of  [default config](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-indentscope.md#default-config).
